pkg/acsengine/engine.go

```diff
@@ -295,15 +295,25 @@ func prepareTemplateFiles(properties *api.Properties) ([]string, string, error)
 //for example: if the target is the public azure, then the default container image url should be gcr.io/google_container/...
 //if the target is azure china, then the default container image should be mirror.azure.cn:5000/google_container/...
 func GetCloudSpecConfig(location string) AzureEnvironmentSpecConfig {
-	cloudSpecConfig := AzureCloudSpec
 	switch location {
-	case "AzureCloud":
-		cloudSpecConfig = AzureCloudSpec
-	case "AzureChinaCloud":
-		cloudSpecConfig = AzureChinaCloudSpec
+	case "chinaeast":
+		fallthrough
+	case "chinanorth":
+		return AzureChinaCloudSpec
+	default:
+		return AzureCloudSpec
 	}
+}
 
-	return cloudSpecConfig
+func GetCloudTargetEnv(location string) string {
+	switch location {
+	case "chinaeast":
```

Comment:
Can we canonicalize them first? Is it true that for any "Location Name" that "locationname" will always be valid?

@weinong is probably right even though I wish he weren't... if that makes sense...